src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIBarbarianAttackMelee.java

```diff
@@ -0,0 +1,107 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import com.minecolonies.api.util.ItemStackUtils;
+import net.minecraft.entity.EntityCreature;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.ai.EntityAIBase;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemSword;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.EnumHand;
+
+/**
+ * Created by Asher on 13/6/17.
+ */
+public class EntityAIBarbarianAttackMelee extends EntityAIBase
+{
+
+    private final EntityCreature   entity;
+    private       EntityLivingBase target;
+    private              int    lastAttack              = 0;
+    private static final int    CYCLES_TO_WAIT          = 16;
+    private static final double PITCH_MULTIPLIER        = 0.4;
+    private static final double PITCH_ADDER             = 0.8;
+    private static final double DEFAULT_DAMAGE          = 3.0;
+    private static final double HALF_ROTATION           = 180;
+    private static final double MIN_DISTANCE_FOR_ATTACK = 2.5;
+    private static final double ATTACK_SPEED            = 1.3;
+    private static final int    MUTEX_BITS              = 3;
+    private static final double IS_ZERO                 = 0;
+
+    public EntityAIBarbarianAttackMelee(final EntityCreature creatureIn)
+    {
+        super();
+        this.entity = creatureIn;
+        this.setMutexBits(MUTEX_BITS);
+    }
+
+    @Override
+    public boolean shouldExecute()
+    {
+        target = entity.getAttackTarget();
+        return target != null;
+    }
+
+    /**
+     * Returns whether an in-progress EntityAIBase should continue executing
+     */
+    public boolean continueExecuting()
+    {
+        if (target.isEntityAlive() && entity.isEntityAlive() && !target.getIsInvulnerable())
+        {
+            attack(target);
+            return true;
+        }
+        return false;
+    }
+
+    public void startExecuting()
+    {
+        attack(target);
+    }
+
+    public void attack(final EntityLivingBase target)
+    {
+        double damageToBeDealt = 0;
+
+        final ItemStack heldItem = entity.getHeldItem(EnumHand.MAIN_HAND);
+
+        if (target != null)
+        {
+            if (ItemStackUtils.doesItemServeAsWeapon(heldItem) && heldItem.getItem() instanceof ItemSword)
+            {
+                damageToBeDealt += ((ItemSword) heldItem.getItem()).getDamageVsEntity();
+            }
+
+
+            if (damageToBeDealt == IS_ZERO)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Equality tests should not be made with floating point values. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1244)